chasm/lib/callback/statemachine_test.go

```diff
@@ -0,0 +1,149 @@
+package callback
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.temporal.io/server/chasm"
+	callbackspb "go.temporal.io/server/chasm/lib/callback/gen/callbackpb/v1"
+	"go.temporal.io/server/common/backoff"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/protobuf/proto"
+)
+
+// testMutableContext is a minimal test helper for capturing tasks
+type testMutableContext struct {
+	*chasm.MockMutableContext
+	tasks []testTask
+}
+
+type testTask struct {
+	component  chasm.Component
+	attributes chasm.TaskAttributes
+	message    any
+}
+
+func newTestMutableContext(t *testing.T) *testMutableContext {
+	ctrl := gomock.NewController(t)
+	mockCtx := chasm.NewMockMutableContext(ctrl)
```

Comment:
You're already implementing the mock here with `testMutableContext`, you don't need this generated mock.

You can do all of this with `testMutableContext`:
```go
type testMutableContext struct {
	chasm.MutableContext
	tasks []testTask
}

func (c *testMutableContext) AddTask(component chasm.Component, attributes chasm.TaskAttributes, payload any) {
	c.tasks = append(c.tasks, testTask{component, attributes, payload})
}
```